### DISCUSSION TITLE
Highlight deprecated function and highlight array

### DISCUSSION THREAD

**Discussion by Skillkiller** [algolia/instantsearch#5748] [Q&A] [answered]
I use Instantsearch js to enable search. For the found entries I want to highlight the corresponding attributes. For this I currently use this function ``instantsearch.highlight({ attribute: 'title', hit })``. I know that this function is deprecated. I am using SvelteKit, what is the replacement option in instantsearch.js? So far I have only found examples for React or Angular. 

Furthermore, Instantsearch.js allows me to search using the ``game?`` attribute. How can I highlight this attribute? ``instantsearch.highlight({ attribute: 'game', hit })`` does not work.

Example response
```json
{
	"4": {
		"id": "321081801",
		"title": "Kleiner Survivor Perk Beginnerguide",
		"game": [
			"Dead by Daylight"
		],
		"_formatted": {
			"id": "321081801",
			"title": "Kleiner Survivor Perk Beginnerguide",
			"game": [
				"Dead by Daylight"
			],
		}
	}
}
```

Thanks a lot

**Comment by Haroenv** [comment]
Where are you using highlight now? If inside a hits widget, you can use the `components` instead:

```js
hits({
  templates: {
    item: (hit, { html, components }) => html`${components.Highlight({ hit, attribute: 'game' })}`
  }
})
```

If you have another use case I'd be interested into how you were using the "string-based" highlight previously, injecting it as "dangerous html" in svelte?

**Reply by Skillkiller** [reply]
I have forgotten this information. I use a custom hits widget. For this I connect the connectHits with my renderHits method. Similar to the example in the [documentation](https://www.algolia.com/doc/api-reference/widgets/hits/js/#click-and-conversion-events).

Here my code strongly shortened:
```ts
// Create the render function
const renderHits = (renderOptions: any, isFirstRender: any) => {
    const { hits, widgetParams } = renderOptions;

    widgetParams.container.innerHTML = ``
    const outerDiv = document.createElement('div');
    outerDiv.className = "containter-fluid";
    widgetParams.container.appendChild(outerDiv);

    hits.forEach((hit: any, index: number) => {
        const innerDiv = document.createElement('div');
        innerDiv.className = "card mb-2 p-2 w-100";
        innerDiv.innerHTML = `...
        <div class="col">
        <img src="/svg/joystick.svg" />${instantsearch.highlight({ attribute: 'game', hit })}
    </div>
    ...
        `;
        outerDiv.appendChild(innerDiv);
    });

    widgetParams.tmp = hits.map((hit: any) => hit.thumbnailUrl);
};

// Create the custom widget
export const bootstrapHits = connectHits(renderHits);
```

How can I display and mark the game array?

**Reply by Haroenv** [reply, marked as answer]
Right, for innerHTML we don't have an alternative yet, although you can do `hit._highlightResult.attribute.value.replace(/__(\/)?ais-highlight__/g, '<$1mark>')` in your case as well, as it's equivalent and not deprecated

**Reply by Skillkiller** [reply]
Thank you for your solution :)
